src/pmi/pmi.cs

```diff
@@ -820,12 +824,21 @@ private List<Type> GetInstances(Type type)
                 }
                 else if (genericArguments.Length == 2)
                 {
-                    newType = type.MakeGenericType(firstType, secondType);
+                    foreach (Type secondType in typesToTry)
+                    {
+                        if (!AreConstraintsSatisfied(secondType, genericArguments[1]))
+                        {
+                            continue;
+                        }
+
+                        newType = type.MakeGenericType(firstType, secondType);
```

Comment:
should there be a try/catch around this, so we try the rest of the "typesToTry" for the second argument?